containers/src/Data/IntMap/Internal.hs

```diff
@@ -43,17 +45,252 @@
 -- This defines the data structures and core (hidden) manipulations
 -- on representations.
 --
+-- = Tree Structure
+--
+-- This implementation uses a novel modification of /big-endian patricia trees/, structured
+-- as a vantage-point tree under the xor metric.
```

Comment:
I'm not entirely sure what to cite for XOR metric. All I mean is viewing the XOR function as a [metric](https://en.wikipedia.org/wiki/Metric_(mathematics)) (i.e. a positive-definite, symmetric function satisfying the triangle inequality). The only major CS work I've seen using it in this way, the [Kademlia](https://www.scs.stanford.edu/~dm/home/papers/kpos.pdf) DHT, is kind of irrelevant.

(I'm also not happy with the prime placement of the VP-tree reference here. I only thought of the connection when trying to better explain why comparing XORs works to navigate the tree, and while I find the analogy especially helpful there, it seems kind of tangential in most cases. A VP-tree is a useful general structure for searching in an arbitrary metric space with possibly many dimensions and no ability to order things at all, while this structure is very tied to linearly-ordered, twos-complement binary integers.)

Reply:
Ah, I wasn't familiar with this definition of "metric". While googling, I noticed that the terms "XOR distance metric" or just "XOR distance" were also used. These terms remind me of the Hamming distance or the euclidean distance, so how about using one of those?

Since there's no good reference for XOR distance metric, I'd suggest that you include a short explanation in the haddocks. Maybe you could re-use the examples from the haddocks for `IntMap_`?!

BTW if "XOR" is the more common capitalization, I'd suggest that you adopt it. That's not very important though.

Reply:
"XOR distance metric" sounds redundant to me. An explanation of the fact that XOR is a metric, or a link to one, should be sufficient and helpful.